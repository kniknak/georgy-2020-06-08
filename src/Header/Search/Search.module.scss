.input {
    display: flex;
    height: 48px;
    width: 100%;
    justify-content: center;
    align-items: center;
    border: 1px solid gray;
    border-radius: 4px;
    padding: 12px;
    box-sizing: border-box;
    outline: none;

    &:focus, &:active {
        outline: none;
        border-color: cornflowerblue;
    }
}

