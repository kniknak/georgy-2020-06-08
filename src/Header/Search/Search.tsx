import React, { useCallback } from "react";
import * as styles from "./Search.module.scss";

export interface SearchProps {
    search: string
    setSearch: (search: string) => void
}

export const Search: React.FC<SearchProps> = ({ setSearch, search }) => {
    const onChange: React.ChangeEventHandler<HTMLInputElement> = useCallback(event => {
        setSearch(event.target.value);
    }, [setSearch]);

    return (
        <input
            className={styles.input}
            type="text"
            value={search}
            onChange={onChange}
            placeholder="Search documents..."
        />
    );
};
