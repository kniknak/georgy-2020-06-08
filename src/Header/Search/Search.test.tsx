import React from "react";
import { shallow } from "enzyme";
import { Search, SearchProps } from "./Search";

describe("Header", () => {
    describe("Search", () => {
        const defaultProps: SearchProps = {
            search: "foo",
            setSearch: () => undefined,
        }

        it("renders", () => {
            const subject = shallow(<Search {...defaultProps}/>);

            expect(subject).toMatchSnapshot();
        });

        it("handles input change", () => {
            const props: SearchProps = {
                ...defaultProps,
                setSearch: jest.fn()
            }

            const subject = shallow(<Search {...props}/>);

            const value = "bar"

            subject.find("input").prop('onChange')!({ target: { value }} as any)

            expect(props.setSearch).toHaveBeenLastCalledWith(value);
        });
    });
});