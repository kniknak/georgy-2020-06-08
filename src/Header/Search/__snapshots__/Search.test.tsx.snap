// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Search renders 1`] = `
<input
  onChange={[Function]}
  placeholder="Search documents..."
  type="text"
  value="foo"
/>
`;
